File: TwoSum.py
class Solution:
    def twoSum(self, nums, target):
        num_indices = {}
        
        for index, num in enumerate(nums):
            complement = target - num
            if complement in num_indices:
                return [num_indices[complement], index]
            num_indices[num] = index
        
        return []

# Example usage
nums = [2, 7, 11, 15]
target = 9

# Create an instance of the Solution class
solution = Solution()
result = solution.twoSum(nums, target)
print(result)  # Output: [0, 1]

File: TwoSum.java
//Leetcode - 1(TwoSum)
import java.util.Hashtable;
import java.util.Arrays;
class Solution{
public int[] twoSum(int [] nums, int target){
Hashtable<Integer,Integer> numbersFound= new Hashtable<>();
for(int i = 0 ; i < nums.length; i++){
if(numbersFound.containsKey(target - nums[i])){
return new int[]{numbersFound.get(target - nums[i]) , i};
}
numbersFound.put(nums[i],i);
}
return null;
} 
}
/*
This is main method to check the given array
class TwoSum{
public static void main(String [] args){
Solution s = new Solution();
int [] nums = {4, 7, 2, 6};
int target = 9;
int [] A = s.twoSum(nums,target);
System.out.println("Output"+ Arrays.toString(A));
}
}
*/
